[clang-format] Format project using rule file

black-dragon74 opened the pull request:
Some of the files are not formatted according to the rules
defined in .clang-format which lead to multiple indentation
related changes when someone opens a PR which had a change
in the unformatted files.

Reviewer note: All the changes are generated by clang-format using `-style=file`

Signed-off-by: black.dragon74 <nickk.2974@gmail.com>

Commit:
server-helpers.c: streamline get_frame_from_request() function

Remove some unneeded calls and checks and loops.
As an example, once RPC_AUTH_ROOT_SQUASH and RPC_AUTH_ALL_SQUASH are
called once, bail out of the loop.
Call gid_resolve() directly if priv->server_manage_gids is true, etc.

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
glusterd: fix dict data leak

Unconditionally free serialized dict data
in '__glusterd_send_svc_configure_req()'.

Found with AddressSanitizer:

==273334==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 89 byte(s) in 1 object(s) allocated from:
    #0 0x7fc2ce2a293f in __interceptor_malloc (/lib64/libasan.so.6+0xae93f)
    #1 0x7fc2cdff9c6c in __gf_malloc libglusterfs/src/mem-pool.c:201
    #2 0x7fc2cdff9c6c in __gf_malloc libglusterfs/src/mem-pool.c:188
    #3 0x7fc2cdf86bde in dict_allocate_and_serialize libglusterfs/src/dict.c:3285
    #4 0x7fc2b8398843 in __glusterd_send_svc_configure_req xlators/mgmt/glusterd/src/glusterd-svc-helper.c:830
    #5 0x7fc2b8399238 in glusterd_attach_svc xlators/mgmt/glusterd/src/glusterd-svc-helper.c:932
    #6 0x7fc2b83a60f1 in glusterd_shdsvc_start xlators/mgmt/glusterd/src/glusterd-shd-svc.c:509
    #7 0x7fc2b83a5124 in glusterd_shdsvc_manager xlators/mgmt/glusterd/src/glusterd-shd-svc.c:335
    #8 0x7fc2b8395364 in glusterd_svcs_manager xlators/mgmt/glusterd/src/glusterd-svc-helper.c:143
    #9 0x7fc2b82e3a6c in glusterd_op_start_volume xlators/mgmt/glusterd/src/glusterd-volume-ops.c:2412
    #10 0x7fc2b835ec5a in gd_mgmt_v3_commit_fn xlators/mgmt/glusterd/src/glusterd-mgmt.c:329
    #11 0x7fc2b8365497 in glusterd_mgmt_v3_commit xlators/mgmt/glusterd/src/glusterd-mgmt.c:1639
    #12 0x7fc2b836ad30 in glusterd_mgmt_v3_initiate_all_phases xlators/mgmt/glusterd/src/glusterd-mgmt.c:2651
    #13 0x7fc2b82d504b in __glusterd_handle_cli_start_volume xlators/mgmt/glusterd/src/glusterd-volume-ops.c:364
    #14 0x7fc2b817465c in glusterd_big_locked_handler xlators/mgmt/glusterd/src/glusterd-handler.c:79
    #15 0x7fc2ce020ff9 in synctask_wrap libglusterfs/src/syncop.c:385
    #16 0x7fc2cd69184f  (/lib64/libc.so.6+0x5784f)

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Fixes: #1000

black-dragon74 committed:
[configure.ac] Match against Darwin version

macOS Dawrin Kernel has a standard versioning which is used by macOS
itself at various places. It is better to use that instead of the
non-standard sw_vers utility as its output format might change. Using
this also helps us avoid extra parsing.

Improves Upon: #3167
Fixes: #3166

Change-Id: I3fe323463d19ce09fd9c06bf2fbefcf89d632720
Signed-off-by: black-dragon74 <niryadav@redhat.com>

black-dragon74 committed:
[socket.c] Enable AES cipher for SSL by default

Almost all of the CPUs in the market have HW Accel for AES enabled by default.
So we can skip the check for it and enable it by default.

Signed-off-by: black.dragon74 <nickk.2974@gmail.com>

Commit:
Fixes:3177

Change-Id: I7ae0283cfc5d02d7168d750cf6354a6c8e3f94f0
Signed-off-by: JamesWSWu <wu.shiwei@zte.com.cn>

black-dragon74 committed:
[clang-format] Format project using rule file

Some of the files are not formatted according to the rules
defined in .clang-format which lead to multiple indentation
related changes when someone opens a PR which had a change
in the unformatted files.

Signed-off-by: black.dragon74 <nickk.2974@gmail.com>

xhernandez commented:
Files inside `contrib` are not written by us but borrowed from other projects, so formatting them doesn't make sense.

For the other cases, most of them make the code less clear IMO. Even the same clang-format definition has produced different results over time (with different versions of clang), so we decided to not enforce the clang-format mismatches to accept a patch. Just add a comment so that the owner and reviewers could easily check if something is critical or not.

Commit:
libglusterfs: cleanup iobufs

Remove useless linkage fields from 'struct iobuf_arena' and
'struct iobuf_pool', drop unused 'iobpool_default_pagesize()'
macro as well as unused arguments of '__iobuf_arena_destroy()'
and '__iobuf_arena_unprune()', add trivial '__iobuf_free()'
and so ensure that iobuf's lock is destroyed on all paths,
adjust related code and comments.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

Commit:
store, glusterd-store: improve store read iteration code (#3044)

- Removed gf_store_validate_key_value() - it wasn't needed, as gf_store_read_and_tokenize()
 already performed the validations for both key and value, ensuring neither is NULL.
- Added an 8K buffer to the iter struct and re-used across calls to gf_store_iter_get_next().
- Used MALLOC instead of CALLOC for the iter structure, as we initialized all its fields.
- Extracted the buffer definition from the loop, where applicable.

 Updates: #1000
 Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mohit84 committed:
dht: Synchronize layout_(ref|unref) during layout_(get|set) in dht co… (#3263)

Currently dht_layout(ref|unref) is happening after increase/decreate
atomic counter. There is some race condition during new layout has been
changed at the same time old layout is using in other thread.Due to this
a client process is getting crashed.

Solution: Remove layout->ref_lock and use inode->lock to synchronize
          layout.

Fixes: #3262
Change-Id: I55952484bd651176f818b9cab54274f37dbca1a6
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

mohit84 committed:
dht: coverity fix CID 1124365

Updates: #1060
Change-Id: I53640a8fd21284a9dcc8dfa98e1833aabdefc3e3
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

Commit:
cli.c: do not use iobuf pool (#2966)

* cli.c: do not use iobuf pool

There's really no need to allocate an iobuf pool for the CLI.
It's a waste of memory and CPU to do all this allocation for CLI command
invocation.

Fixes: #2963
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

* iobuf.c: make iobuf_get_from_small() non static and use in cli

Re-use the function, by making it non-static.

(Remove the unneeded use of 'ret' in the function while at it).

Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
event-epoll.c: create the 1st table on event epoll creation (#3283)

* event-epoll.c: create the 1st table when creating a new epoll

Instead of later, under lock, create it sooner. It's always going to be used.

Updates: #3228
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

* event-epoll.c: do not re-take a lock when allocating

Reuse the locking done earlier to allocate the slot.

Updates: #3228
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

rafikc30 committed:
afr/metadata-heal: Avoid healing mdata for Arbiter node (#3258)

It is expected to have differences in time attributes for
Arbiter node as we are not doing most of the operations
on Arbiter node. Hence we don't need to check for mdata
value in the dict while calculating sinks for a metadata
heal.

Change-Id: I326017fcb5810d7ce765e189167a4101c6b0a046
Signed-off-by: Mohammed Rafi KC <rafi.kavungal@iternity.com>

tshacked committed:
build: deploy /usr/lib/tmpfiles.d/gluster.conf during client installation (#3233)

The config file for creating /run/gluster on boot is available only with the glusterfs-server package.
As a result on a machine with only client installation, the path isn't created after reboot.
Install the conf file also during client installation.

Fixes:	#3232
Change-Id: Idbbc209b1bd0e82286eb9160c103bf0755cb90b9
Signed-off-by: Tamar Shacked <tshacked@redhat.com>

mohit84 committed:
nfs: Optimize ctxcount value to reduce memory consumption for nfs-server (#3237)

In case of gnfs for nfs-server all volumes are attached with a single graph.
During inode table creation it sets ctxcount same to the total xlators are
attached with a graph. If a user has created 100 volumes (1x3) so total
number of xlators are attached with graph is 702 per volumes almost 7
xlators are attached (io-stat, dht, write-behind, afr, c1, c2, c3) so total
ctxcount set is 702. During inode creation we do allocate memory based on
number of ctxcount to save the xlator data on inode. In case of 100 volumes
memory allocation per inode will be around 32(bytes per ctxcount) * 702 = 22464 Bytes
for ctx.In caseof gnfs default lru_limit is 90000 so total memory allocation will
be 22464*90000(almost 2G). Per volume memory consumption is 2G so it will be huge
for all 100 volumes if every volumes has reached till maximum lru_limit.

Solution: An inode is associated with a specific volume so it is not
          possible multiple volume can save the data on inode_ctx so
          set the ctxcount based on the xlators associated with a volume
          not all the volumes.With this change It will reduce huge
          memory consumption for nfs-server. After set ctxcount (702/100 + 2 = 9)
          total memory allocation will be 32 * 9 * 90000 = 26M per volumes
          it is a huge saving and even for all 100 volumes has reached at
          maximum lru_limit memory consumption will not be more than 2.6G.

Change-Id: If35804d4337723ea902a471ca28842a595030e78
Fixes: #3236
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

Commit:
rpc: cleanup data structures once again

Drop unused 'rpc_transport_data_t' as well as a few unused
members of 'struct rpc_req' and 'struct rpc_transport_req',
adjust related code.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

Commit:
readdir-ahead.c: few improvements

- Used a pointer to an iatt struct instead of a copy.
- Moved some code that was unconditionally used to be used conditionally.
- Removed a memset() to a guid that was just before a call to uuid_utoa_r()
- Reorder rda_fd_ctx struct to reduce padding.

Updates: #1000
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mohit84 committed:
dht: Remove dead code function dht_nonblocking_inodelk

dht_nonblocking_inodelk(_cbk) is not using anywhere so
remove the functions from dht codebase.

Fixes: #3294
Change-Id: Iefe0bf2d7bb4cc622858f80777d5c693e04c733b
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

Commit:
rpc clnt - cleanups (#3146)

1. Ensure xid is uint32_t everywhere
2. Remove unused parameters (for example, rpc_clnt_reply_deinit() does not use
the memory pool parameter, rpc_clnt_reply_init() does not need the saved_frame
parameter)
3. Make all relevant functions static
4. rpc_clnt_notify() was setting old_THIS to THIS twice, removed one.
5. Other code simplifications, mainly for readability.

This is all in preparation to making __saved_frame_copy() more efficient:
There is no need to copy the whole frame there, we just need the rpcreq and the
rsp structures.

There were two structs that were either zero'ed way before they may get used (auth data)
and were zero'ed twice - one on initialization, the other with memset().

Remove 'THIS' call from __saved_frames_put(). Instead, we can use rpc_clnt->owner

__saved_frame_copy() can be more efficient:
There is no need to copy the whole frame there, we just need the rpcreq and the
rsp structures.

rpc_clnt_record_build_record() and rpc_clnt_record_build_header()
now just uses the existing iovec header instead of copying its values around.

In addition, make all relevant functions in the file static, if applicable.

Updates: #1000
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
rpc-clnt.c: properly copy saved_frame_rsp

Thanks to Coverity warning on overrun.

Updates: #3304
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

anoopcs9 committed:
gfapi: Fix return from glfs_open() to honour O_DIRECTORY flag

As per POSIX[1] specifications honour O_DIRECTORY flag during open as
follows:

O_DIRECTORY
    If path resolves to a non-directory file, fail and set errno to
    [ENOTDIR].

ERRORS

* [EISDIR]
  The named file is a directory and oflag includes O_WRONLY or O_RDWR,
  or includes O_CREAT without O_DIRECTORY
* [ENOTDIR]
  A component of the path prefix names an existing file that is neither
  a directory nor a symbolic link to a directory; or O_CREAT and O_EXCL
  are not specified, the path argument contains at least one non-<slash>
  character and ends with one or more trailing <slash> characters, and
  the last pathname component names an existing file that is neither a
  directory nor a symbolic link to a directory; or O_DIRECTORY was
  specified and the path argument resolves to a non-directory file.

The presence of O_DIRECTORY is to ensure the file being opened is a
directory.

[1] https://pubs.opengroup.org/onlinepubs/9699919799/functions/open.html

Signed-off-by: Anoop C S <anoopcs@cryptolab.net>

Commit:
bit-rot: simplify and cleanup time management

Drop trivial 'br_fsscan_calculate_delta()', consistently use
'unsigned long' and 'time_t' to represent time intervals here
and there, drop unused 'tv' from 'struct br_child', adjust
related code.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

pbhatia22 committed:
trash.c: coverity fix (#3205)

Issue: Dereference after Null check

Description: When Pathbuf is NULL passed at multiple points for strncpy
and strncmp producing an error so instead of checking pathbuf is NULL
at multiple points check when pathbuf is updated in check_pathbuf()
function so no need to check at others and also move common code in
single function.

Issue: #1060
CID:1469496, 1469497,1469498
Change-Id: I1791e5ad60f0bb0b1056efd248e515921a007006
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

Commit:
glusterd: drop unused pointer to private data

Massively remove 'glusterd_conf_t *' pointer where it's
actually used for 'GF_ASSERT()' only since there is no
point in blindly checking the same pointer again and
again even in debug configuration.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

black-dragon74 committed:
[event-epoll.c] Check for slots in next table if all slots are used (#3268)

If all epoll slots are used in the current table look for the free slots
in the next table. In rare cases, if a gluster process required more than
1024 slots the previous code returned -1. This patch addresses the same.

Signed-off-by: black-dragon74 <niryadav@redhat.com>

Commit:
multiple files: improve gf_dirent_for_name() functionality (#3008)

- Introduced gf_dirent_for_name2(), which populates most of the fields
with real content and replaced gf_dirent_for_name() in several cases.
- Reduced the no. of calls for strlen() across the code, re-used existing
length fields where applicable.
- Removed dead code
- improve define
- use MALLOC instead of CALLOC (and initialize relevant fields, so everything is init'ed).

Updates: #3005
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

black-dragon74 committed:
[posix.c] Coverity: Handle forward NULL

This patch addresses the coverity forward null error.

CID: 14760{19,20,21}

Signed-off-by: black.dragon74 <nickk.2974@gmail.com>

pbhatia22 committed:
fix for warnings (#3290)

mohit84 committed:
server:Optimize server_process_event_upcall code path

The function server_process_event_upcall handles upcall event and send a
notification to client. To send a notification to client it has taken
a lock to access xprt and the lock has not released until the event has not
been sent to the client, It means at a time it will not able to process
more than one notification it has lock contention while a brick process is trying
to send more than one upcall event notification.

Solution: Instead of sending an event under critical section take a reference
          on the xprt.
Fixes: #3321
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

Change-Id: Ibaab1626ab6e0c318df77e8efe528e71f3802517

black-dragon74 commented:
@xhernandez Should I just close this then?

Commit:
glusterd: fix option flags

Fix (presumably typo) in 'performance.aggregate-size'
and 'features.tag-namespaces' option flags.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

xhernandez commented:
> @xhernandez Should I just close this then?

I think we shouldn't apply these massive formatting changes, but let's see what others think... @amarts @pranithk @mohit84

black-dragon74 commented:
> I think we shouldn't apply these massive formatting changes

Agreed.

chen1195585098 committed:
socket: fix address family resolution (#3339)

Defective code can be found in function
af_inet_client_get_remote_sockaddr :

if (inet_pton(AF_INET6, remote_host, &serveraddr_ipv6)) {
    sockaddr->sa_family  = AF_INET6;
}

The problem is, when a ipv6 hostname comes here, it fails
to pass the condition. Hence, the sa_family of this ipv6
hostname cannnot be correctly marked as AF_INET6, but
AF_INET.

it can be verified by following steps:
1. create volume in ipv6 (ip or hostname)
2. mount the volume using hostname
3. mount failed, since the hostname is incorrectly
   set to AF_INET as displayed in log file.

Fixes: #3329
Change-Id: Ibd1b9e71f2d48a205eef1ce3cebd7397ce26e94f
Signed-off-by: ChenJinhao <jinhaochen.cloud@gmail.com>

black-dragon74 committed:
[inode.c] Optimize setting `xl_key` on inode's ctx (#3207)

`xl_key` on an inode's ctx is already set to `xlator` when index > 0
inside `__inode_get_xl_index`. There is no need to set it again after
getting the index.

Change-Id: I606307653c47f3a57cfe355e290468f3e68e0d33
Signed-off-by: black-dragon74 <niryadav@redhat.com>

Commit:
event-epoll.c: do not CALLOC a whole table (#3285)

Since we memset() each slot when we allocate it, no need to CALLOC() the
whole table (1K slots * 120 bytes each!)

For some reason, every slot had a 'slots_used' which was wasteful, as this is
a per-table value, not a per slot. Re-arranged the code to have a table structure
with a table_slots_used variable to store how many slots used.

(In the future, I think it makes sense to move it to an 64bit bitmap and have 64 entries per table, instead of 1024)

While at it, re-arranged __event_slot_alloc() code a bit.

Updates: #3228
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mohit84 committed:
rpc: avoid restarting the bail out timer after initiating shut down (#3340)

glfs_fini takes time (20s) and slows down qemu-img, the time
is taken by due to timer_thread behavior. The timer thread
reregister function unless the timer event has not been
removed from a list.To avoid the delay the even needs
to add only if cleanup has not started.In case if cleanup
has started there is no need to add the event again in
timer thread list.

Fixes: #3334
Change-Id: I179c63c656c2917de9eb2bb44664453a1b6fc471
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

Commit:
socket: more cleanups once again

Drop unused 'GF_REF_DECL' and convert 'ssl_meth' and 'ssl_sbio'
of 'socket_private_t' / 'iobuf_size' of 'struct gf_sock_incoming'
to local variables, omit trivial 'socket_proto_state_machine()'
wrapper, adjust related code and simplify '__socket_reset_priv()'
to '__socket_reset_incoming()'.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

amarts commented:
Can you see what is the impact if you don't do changes inside `contrib/` and `tests/` ? those 2 directories were intentionally left out as people may have to do a diff in different project to make improvements later. If we apply our coding standard there, then its not easy to do diff in those particular projects.

If the diff is small after that, then we can still consider this.

Commit:
afr: cleanup callback continuation (#3280)

Drop 'afr_local_t' members which aren't actually
used to serve AFR fop callbacks, adjust comments.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

Commit:
dht-helper.c: simplify dht_local_init() (#3293)

* dht-helper.c: simplify dht_local_init()

1. Remove a call to 'THIS' and use frame->this instead
2. Re-use the previous result of dht_layout_get() to fetch the cached_subvol.

Updates: #1000
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

* Fix review comments

Updates: #1000
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
meta.c/.h: meta_lookup() and other small improvements (#3360)

The meta_lookup() function would use two macros:
META_HOOK - which would call THIS to fetch this->private
IS_META_ROOT_GFID - which did a strcmp() to the result of a uuid_utoa() function.

Simplified by saving the result of the uuid_utoa() in the private struct, doing the comparison directly, and without the call to 'THIS'.
In turn, re-used that result to copy that UUID instead of parsing it (again).

Found out you can skip most of this if 'loc->name' is null, so added it as well.

Updates: #3359
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
loc_gfid and loc_pargfid should not unconditionally clear the gfid (#3361)

There was no reason to unconditionally clear the gfid.
Now we clear it only when needed.

Updates: #1000
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
Upcall minor improvements (#3318)

It was already called only if upcall was enabled, in all places but one.
Added that one (up_writev_cbk) as well and removed the check from
upcall_cache_invalidate() which was useful in case where it is called in a loop,
for example in up_readdirp_cbk()

In addition, wherever it was called with a constant flag, used that constant.

Updates: #1000
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
dht: do not print uuid in binary form (#3364)

Fix 'dht_protect_namespace()' and 'dht_selfheal_dir_mkdir()'
by using 'uuid_utoa()' to avoid printing uuid in binary form.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

amarts committed:
Simple quota: based on namespace (#1750)


* the quota setting can happen only on 'namespace' inode.
* once set, the accounting is maintained only at the namespace level for whole tree.
* uses 'simple-quota' key to show the correct quota usage in distributed volume.
* statfs()'s response would be used to set the the volume level usage in xlator,
  in setxattr() call, when done through a special mount process.
* the xlator is designed to be on brick graph, and saves only the usage of data
  inside it.
* An option is provided to utilize the backend filesystem's quota feature for
  'accounting', which can improve the performance.
  - This PR expects backend quota to properly return `statfs()` (or `df`) output.
    If `features.simple-quota.use-backend` option is set, then there wont be any
    active accounting in simple-quota translator, but only `setxattr()` and
    `statfs()` is handled for special keys. We expect helper function to
    set 'quota-limit' and also 'namespace' xattr to aid the glusterfs process in
    general to identify the entries.

Updates: #1774
Change-Id: Id4229b720b57cde458b9b36e36ada3ffe2be0ac2
Signed-off-by: Amar Tumballi <amar@kadalu.io>

harshita-shree committed:
fix compile warning[glusterd-store] (#3370)

Warning:

glusterd-store.c:4355:36: warning: ‘snprintf’ output may be truncated before the last format character [-Wformat-truncation=] 4355 | snprintf(peerfpath, len, "%s/%s", peerdir, str); | ^ glusterd-store.c:4355:5: note: ‘snprintf’ output 2 or more bytes (assuming 4097) into a destination of size 4096 4355 | snprintf(peerfpath, len, "%s/%s", peerdir, str);

Description: Added check for overflowing warning.

Updates : #1000
Change-Id: I7b7b95ab3a71d7cf6b7d2143fea81b819e3b34d8
Signed-off-by: Harshita Shree <hshree@redhat.com>

pbhatia22 committed:
glusterd-snapd-svc.c: fix for compile warning (#3363)

Warning:

glusterd-snapd-svc.c:37:5: note: ‘snprintf’ output between 11 and 4106 bytes into a destination of size 4096
     snprintf(logfile, len, "%s/snapd.log", logdir);

Description: As both logfile and logdir having same size so this warning arises so added
extra check for this warning

Updates : #1000
Change-Id: I05d5905c192936d3f52f8813f94baac929159914
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

harshita-shree committed:
make functions static/remove redundant checks [afr] (#3348)

Made relevant functions static in afr component to improve code readability and performance.
Also removed redundant null checks

Reference to issue #2967
Change-Id: I96172331f1aa0ff2d65ea2a876303ec799bf5144
Signed-off-by: Harshita Shree <hshree@redhat.com>

mohit84 committed:
glusterd: After node reboot not able to start all bricks successfully

The glusterd is not able to start all bricks successfully after node reboot
if brick count is high(>750).The glusterd has attempted all the bricks to
start but brick process is not able to get a response from glusterd because glusterd
is busy to start the volumes so it has disconnect with glusterd.

Solution: Start the volumes with wait flag to true so that at the time
          of making a connection with brick process the brick process
          has started successfully.

Fixes: #3375

Note: To validate the patch follow the steps
1) Setup 400 1x3 volumes on a single node(without brick_mux)
2) Start all the volumes
3) kill all the gluster processes
    pkill -f gluster
4) Start glusterd
5) No volume should throw "disconnected from glusterd"

Change-Id: Id050153c581dd74539cb0a8e65270adf216c9b26
Signed-off-by: Mohit Agrawal <moagrawa@redhat.com>

Commit:
dict.c/h: remove hash code

We don't use it anymore, we can remove it altogether.

Updates: #3382
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

pbhatia22 committed:
fix_warnings

pbhatia22 committed:
graph.c: fix for compile warning (#3369)

Warning:

graph.c: In function ‘fill_uuid.constprop’:
graph.c:537:33: warning: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size 125 [-Wformat-truncation=]
     snprintf(uuid, size, "%s-%d-%s", gf_gethostname(), getpid(), now_str);
                                 ^~                               ~~~~~~~
graph.c:537:5: note: ‘snprintf’ output 4 or more bytes (assuming 259) into a destination of size 128
     snprintf(uuid, size, "%s-%d-%s", gf_gethostname(), getpid(), now_str);

Description: Warning arises as destination buffer having less size
so added extra check to remove this warning

Updates: #1000
Change-Id: I78d6df65e2b34891cfc6d36d814f25390a279977
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

Commit:
glusterd: cleanup gsyncd running code (#3368)

Prefer 'run_gsyncd_cmd()' over (somewhat weird)
RUN_GSYNCD_CMD macro, adjust related code.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

Commit:
make functions static/remove redundant checks [afr] (#3348) (#3327)

Made relevant functions static in afr component to improve code readability and performance.
Also removed redundant null checks

Reference to issue #2967
Change-Id: I96172331f1aa0ff2d65ea2a876303ec799bf5144
Signed-off-by: Harshita Shree <hshree@redhat.com>
Co-authored-by: Harshita Shree <97235968+harshita-shree@users.noreply.github.com>

amarts committed:
O_PATH support over protocol (#3385)

This doesn't break XDR as the flag is passed as 32bit number.
But the support for O_PATH will come only when both the client
and server have this code in it to translate the number over wire.

Change-Id: I057eefe4bb0f93f27cd844268ffca21254bd7fbf
Signed-off-by: Amar Tumballi <amar@kadalu.io>

pbhatia22 committed:
fix for warnings (#3392)

pbhatia22 committed:
clang issue fix (#3390)

Issue: Access to field 'open_count' results in a dereference of
an undefined pointer value (loaded from variable 'ob_inode')

Fix: First initialized ob_inode and then check for ob_inode if it is NULL

Updates: #1060
Change-Id: I26e5873e2a73613fcce127f3da81751121f55b79
Signed-off-by: Preet Bhatia <pbhatia@redhat.com>

xhernandez committed:
xlator_t: remove inconsistent alignment attribute

The 'stats' array inside 'xlator_t' was explicitly aligned to 128 bytes
(CAA_CACHE_LINE_SIZE = 128). Since this is a structure definition, it
caused the whole structure to be considered aligned to 128. That's not
problematic by itself as long as we really allocate the memory blocks
for 'xlator_t' objects at addresses multiple of 128 bytes, because the
compiler may use this fact to generate aligned access instructions
which are faster than unaligned ones.

Since we are not really allocating 'xlator_t' objects at that alignment,
the assumption that the compiler does is wrong, causing a crash.

Fixes: #3417
Change-Id: I33b5021035aae24f0b5ea14e255873da79fc0625
Signed-off-by: Xavi Hernandez <xhernandez@redhat.com>

xhernandez committed:
logging: new way of defining and using log messages (#3342)

Messages from quiesce xlator have been migrated to the new format as an
example.

This new approach has several advantages:

  - Centralized message definition
  - Customizable list of additional data per message
  - Typed message arguments to enforce correctness
  - Consistency between different instances of the same message
  - Better uniformity in data type representation
  - Compile-time generation of messages
  - Easily modify the message and update all references
  - All argument preparation is done only if the message will be logged
  - Very easy to use
  - Code auto-completion friendly
  - All extra overhead is optimally optimized by gcc/clang

The main drawback is that it makes heavy use of some macros, which is
considered a bad practice sometimes, and it uses specific gcc/clang
extensions, but we are already using them in other places.

To create a new message:

    GLFS_NEW(_comp, _name, _msg, _num, _fields...)

To deprecate an existing message:

    GLFS_OLD(_comp, _name, _msg, _num, _fields...)

To permanently remove a message (but keep its ID reserved):

    GLFS_GONE(_comp, _name, _msg, _num, _fields...)

To be able to mix messages using old and new interfaces, the messages
using the old interface can be defined this way:

    GLFS_MIG(_comp, _name, _msg, _num, _fields...)

Each field is a list composed of the following elements:

    (_type, _name, _source, _format, (_values) [, _extra])

 - _type   is the data type of the field (int32_t, const char *, ...)
 - _name   is the name of the field (it will also appear in the log
           string)
 - _source is the origin of the data
 - _format is the C format string to represent the field
 - _values is a list of values used by _format (generally it's only
           _name)
 - _extra  is an optional extra code to prepare the representation of
           the field (check GLFS_UUID() for an example)

There are some predefined macros for common data types.

Example:

Message definition:

    GLFS_NEW(LIBGLUSTERFS, LG_MSG_EXAMPLE, "Example message", 3,
        GLFS_UINT(number),
        GLFS_ERR(error),
        GLFS_STR(name)
    )

Message invocation:

    GF_LOG_I("test", LG_MSG_EXAMPLE(3, -2, "test"));

This will generate a message like this:

    "Example message ({number=3}, {error=2 (File not found)}, {name='test'})"

Debug and trace messages are defined directly in the logging place:

    GF_LOG_D("test", "Debug message", 3,
        GLFS_UINT(number, 3),
        GLFS_ERR(error, errno),
        GLFS_STR(name, this->name)
    );

Change-Id: I8f4bd7b9b90f649a52fe29a62222101eeccf0c68
Signed-off-by: Xavi Hernandez <xhernandez@redhat.com>

rafikc30 committed:
afr/heal: avoid automatic gfid split-brain heal of directories (#3289)

When we have more than one distribution set, it may not be
a good idea to automatic gfid split brain resolution using
favourite child policy

Because there could be a chance that the gfid that healed
could be a wrong one, since the source of truth can only
be determined by DHT.

It is still possible to resolve the gfid mismatch using the
CLI command

Change-Id: I4ddc82e6b93da9b8debb74222aff2d7d276673b3
fixes: #3288
Signed-off-by: Mohammed Rafi KC <rafi.kavungal@iternity.com>

karthik-us committed:
[coverity]: Fix resource leak on multiple files (#3404)

CIDs: 1487707, 1487704, 1487702, 1487690, 1487687
Also corrected the return values in glusterd_read_status_file().

Change-Id: Iba03d676dc020ce633ab089f31b4c86fd8aceb1e
Updates: #1060
Signed-off-by: karthik-us <ksubrahm@redhat.com>

black-dragon74 committed:
[clang-format] Format the project and add config

to ignore formatting on contrib and tests dirs.

Signed-off-by: black-dragon74 <niryadav@redhat.com>

black-dragon74 commented:
@amarts I did the requested changes and the changes now are not that massive. You can check the new PR #3431

Had to close this PR due to branch divergence on my end. Sorry for that.
